python/pyspark/errors/utils.py

```diff
@@ -268,6 +266,8 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
                 finally:
                     set_current_origin(None, None)
             else:
+                spark = SparkSession.getActiveSession()
```

Comment:
`getActiveSession` is pretty expensive so we'd better call it only when `is_debugging_enabled()` is True.

Reply:
yeah, we should always invoke SparkSession operations lazily